src/VGrid.cu

```diff
@@ -0,0 +1,554 @@
+/*
+ * Copyright (c) 2018-2020, The Bifrost Authors. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * * Redistributions of source code must retain the above copyright
+ *   notice, this list of conditions and the following disclaimer.
+ * * Redistributions in binary form must reproduce the above copyright
+ *   notice, this list of conditions and the following disclaimer in the
+ *   documentation and/or other materials provided with the distribution.
+ * * Neither the name of The Bifrost Authors nor the names of its
+ *   contributors may be used to endorse or promote products derived
+ *   from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS ``AS IS'' AND ANY
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+ * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+ * EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+ * PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+ * PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+ * OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/* 
+
+Implements the Romein convolutional algorithm onto a GPU using CUDA. 
+
+*/
+#include <iostream>
+#include <bifrost/VGrid.h>
+#include "romein_kernels.cuh"
+
+#include "assert.hpp"
+#include "trace.hpp"
+#include "utils.hpp"
+#include "cuda.hpp"
+#include "cuda/stream.hpp"
+#include "Complex.hpp"
+
+#define MAX_THREADS_PER_BLOCK 128
+#define MIN_BLOCKS_PER_MP     4
+
+struct __attribute__((aligned(1))) nibble2 {
+    // Yikes!  This is dicey since the packing order is implementation dependent!  
```

Comment:
Yes and no.  Yes, we should address it since in Bifrost changing the endianess of a `ci4` values changes the ordering of real and complex within the byte.  No, we shouldn't since I don't know that the other endianess is ever used for LWA and putting in the Bifrost endianess check would be kind of a pain.